api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -1,18 +1,242 @@
+  getCurriculumAssessmentById,
+  getProgramIdByProgramAssessmentId,
+  getAssessmentSubmissions,
+  getFacilitatorAssessmentSubmissionsSummary,
+} from '../../services/assessmentService';
+
+jest.mock('../../services/assessmentService');
+
+const mockPrincipalEnrolledPrograms = jest.mocked(principalEnrolledPrograms);
+const mockGetProgramIdByProgramAssessmentId = jest.mocked(
+  getProgramIdByProgramAssessmentId
+);
+const mockFindRoleInProgram = jest.mocked(findRoleInProgram);
+const mockGetAssessmentsForProgram = jest.mocked(getAssessmentsForProgram);
+const mockGetCurriculumAssessmentById = jest.mocked(
+  getCurriculumAssessmentById
+);
+const mockGetAssessmentSubmissionsSummary = jest.mocked(
+  getAssessmentSubmissions
+);
+const mockGetFacilitatorAssessmentSubmissionsSummary = jest.mocked(
+  getFacilitatorAssessmentSubmissionsSummary
+);
+
+const programAssessmentId = 1;
+const curriculumAssessmentId = 1;
+const programId = 2;
+const facilitatorId = 1;
+const participantId = 2;
+const curriculumAssessment: CurriculumAssessment = {
+  id: curriculumAssessmentId,
+  title: 'Assignment 1: React',
+  description: 'Your assignment for week 1 learning.',
+  max_score: 10,
+  max_num_submissions: 3,
+  time_limit: 120,
+  curriculum_id: 3,
+  activity_id: 97,
+  principal_id: facilitatorId || participantId,
+};
+
+const programAssessment: ProgramAssessment = {
+  id: programAssessmentId,
+  program_id: 1,
+  assessment_id: curriculumAssessmentId,
+  available_after: '2023-02-06',
+  due_date: '2023-02-10',
+};
+
+const assessmentSubmissionsSummary: AssessmentSubmissionsSummary = {
+  principal_id: participantId,
+  highest_state: 'Graded',
+  most_recent_submitted_date: '2023-02-09 13:23:45',
+  total_num_submissions: 1,
+  highest_score: 10,
+};
+
+const facilitatorAssessmentSubmissionsSummary: FacilitatorAssessmentSubmissionsSummary =
+  {
+    num_participants_with_submissions: 6,
+    num_program_participants: 12,
+    num_ungraded_submissions: 2,
+  };
+
+const emptyAssessmentsSummaryList: AssessmentSummary[] = [];
 
 describe('assessmentsRouter', () => {
   const appAgent = createAppAgentForRouter(assessmentsRouter);
   const exampleAssessmentId = 1;
   const exampleSubmissionId = 1;
 
   describe('GET /', () => {
-    it('should respond with a list of all assessments', done => {
-      const response = { behaviour: 'Shows a list of all assessments' };
-      appAgent.get('/').expect(200, itemEnvelope(response), err => {
-        done(err);
-      });
+    const unenrolledPrincipalId = 4;
+    const enrolledParticipantPrincipalId = 5;
+    const facilitatorPrincipalId = 6;
+    const programIds: number[] = [];
+
+    it('should respond with an empty list for a user not enrolled in any programs', done => {
+      // mock response from function that gets a list of programs the user is enrolled in to return an empty list
+      mockPrincipalEnrolledPrograms.mockResolvedValue(programIds);
+
+      appAgent
+        .get('/')
```

Comment:
It would help if you could be more specific on the list of what we are getting here. Are we trying to get a list of programs the mentee is enrolled in? Also, I think that the get '/' path will need to be updated for the path we are trying to go to.